src/blob/index/core.rs

```diff
@@ -55,20 +55,20 @@ where
 
 #[derive(Debug, Default)] // Default can be used to initialize structure with 0
 pub(crate) struct MemoryAttrs {
-    pub(crate) key_size: usize,
-    pub(crate) btree_entry_size: usize,
-    pub(crate) records_allocated: usize,
+    pub(crate) key_size: AtomicUsize,
+    pub(crate) btree_entry_size: AtomicUsize,
     // contains actual size occupied by record header in RAM (which helps
     // to compute actual size of indices in RAM in `InMemory` state)
-    pub(crate) record_header_size: usize,
-    pub(crate) records_count: usize,
+    pub(crate) record_header_size: AtomicUsize,
+    pub(crate) records_count: AtomicUsize,
+    pub(crate) records_allocated: AtomicUsize,
 }
 
 pub type InMemoryIndex<K> = BTreeMap<K, Vec<RecordHeader>>;
 
 #[derive(Debug, Clone)]
 pub(crate) enum State<FileIndex, K> {
-    InMemory(InMemoryIndex<K>),
+    InMemory(Arc<SRwLock<Option<InMemoryIndex<K>>>>),
```

Comment:
`Arc` and `Option` are not needed. They only give overhead. `Clone` on the `State` can be removed

Reply:
Indeed, `mem::take` can be used!